drivers/imx-enet.c

```diff
@@ -704,24 +815,130 @@ static int enet_initMDIO(enet_state_t *state)
 }
 
 
+/*
+ * Enable clock for the required ENET module
+ */
 static int enet_clockEnable(enet_state_t *state)
 {
-	static const platformctl_t pctl_enet_clock = {
-		pctl_set, pctl_devclock, .devclock = { pctl_clk_enet, 3 }
+#if defined(__CPU_IMXRT106X)
+	static const platformctl_t pctl_enet_clock[] = {
+		{ pctl_set, pctl_devclock,
+			.devclock = { pctl_clk_enet, clk_state_run_wait } },
+
+		{ pctl_set, pctl_devclock,
+			.devclock = { pctl_clk_gpio1, clk_state_run_wait } },
 	};
-	int err;
 
-	err = platformctl_seq(&pctl_enet_clock, 1);
-	if (err < 0) {
-		enet_printf(state, "Can't enable ENET clock\n");
+	if (state->dev_phys_addr != ENET_ADDR_ENET1) {
+		enet_warnUnsupportedDeviceAddr(state);
+		return -1;
 	}
 
+	/* RM 14.8.14 */
+#define REG(addr) (*((volatile uint32_t *)(addr)))
```

Comment:
there is no public `platformctl` for this. ENET clock is deinitialized in `phoenix-rtos-kernel/hal/(...)/imxrt10xx` - [595](https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/595) fixes this, so it would need to be merged before this